ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -784,3 +784,90 @@ def test_ipa_certs_check_ipacertnsstrust(self):
                 check["kw"]["key"] in DEFAULT_PKI_KRA_CERTS
             )
         tasks.uninstall_master(self.master)
+
+
+class TestIpaHealthCheckWithADtrust(IntegrationTest):
+    """
+    Test for ipa-healthcheck tool with IPA Master with trust setup
+    with AD system
+    """
+    topology = "line"
+    num_ad_domains = 1
+
+    @classmethod
+    def install(cls, mh):
+        tasks.install_master(cls.master, setup_dns=True)
+        cls.ad = cls.ads[0]
+        tasks.install_adtrust(cls.master)
+        tasks.configure_dns_for_trust(cls.master, cls.ad)
+        tasks.establish_trust_with_ad(cls.master, cls.ad.domain.name)
+
+    def test_ipahealthcheck_trust_domainscheck(self):
+        """
+        This testcase checks when trust between IPA-AD is established,
+        IPATrustDomainsCheck displays result as SUCCESS and also
+        displays ADREALM as sssd/trust domains
+        """
+        returncode, data = run_healthcheck(
+            self.master, "ipahealthcheck.ipa.trust", "IPATrustDomainsCheck"
+        )
+        assert returncode == 0
+        for check in data:
+            if check["kw"]["key"] == "domain-list":
+                assert check["result"] == "SUCCESS"
+                assert (
+                    check["kw"]["sssd_domains"] == self.ad.domain.name
+                    or check["kw"]["trust_domains"] == self.ad.domain.name
```

Comment:
> Not sure about this comment, but shouldn't it be an 'and' instead of an 'or'?

@flo-renaud 
1. Replaced or with and statement.
2. Also added tests for IPATrustAgentMemberCheck, IPATrustCatalogCheck and IPADomainCheck related test.
3. Will be adding more tests for no trust scenario/other checks in seperate PR.